src/main/java/TaskList.java

```diff
@@ -0,0 +1,81 @@
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class TaskList {
+    ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> loadedData) {
+        this.taskList = loadedData;
+    }
+
+    public int size() {
+        return taskList.size();
+    }
+
+    public Task get(int index) {
+        return taskList.get(index);
+    }
+
+    public void add(Task task) {
+        taskList.add(task);
+    }
+
+    public void setDone(int index) {
+        Task newTask = this.taskList.get(index - 1).markAsDone();
+        this.taskList.set(index - 1, newTask);
+        Ui.printMarkAsDone(newTask);
+    }
+
+    public void setDelete(int index) {
+        Task removed = this.taskList.get(index - 1);
+        this.taskList.remove(index - 1);
+        Ui.printRemoveTask(removed, this.taskList);
+    }
+
+    public void setTodo(String input) {
+        String description = input.substring(4);
+        Task newTask = new Todo(description.trim());
+        this.taskList.add(newTask);
+        Ui.printTaskAdded(newTask, taskList);
+    }
+
+    public void setDeadline(String description, LocalDate d1, String timeString) {
```

Comment:
Perhaps a more descriptive variable name for LocalDate instead of d1 could be used? (eg localDate) Noted this issue in several other places too